modules/gapi/src/backends/ov/govbackend.cpp

```diff
@@ -0,0 +1,982 @@
+                const auto &matdesc = cv::util::get<cv::GMatDesc>(mm);
+
+                const auto &input_name = std::get<0>(it);
+                auto &input_info = ppp.input(input_name);
+                input_info.tensor().set_element_type(toOV(matdesc.depth));
+
+                const auto explicit_in_model_layout = lookUp(input_model_layout , input_name);
+                if (explicit_in_model_layout) {
+                    input_info.model().set_layout(::ov::Layout(*explicit_in_model_layout));
+                }
+
+                const auto explicit_in_tensor_layout = lookUp(input_tensor_layout, input_name);
+                if (explicit_in_tensor_layout) {
+                    input_info.model().set_layout(::ov::Layout(*explicit_in_tensor_layout));
+                }
+
+                // NB: Image case - all necessary preprocessng:
+                // layout conversion, resize is configured automatically.
+                // FIXME: What if user provided tensor layout that isn't equal to NHWC?
+                // Current solution just disable all automatic preprocessing...
+                const auto &input_shape = uu.model->input(input_name).get_shape();
+                if (isImage(matdesc, input_shape) &&
+                    ((!explicit_in_tensor_layout) ||
+                     (explicit_in_tensor_layout && (*explicit_in_tensor_layout == "NHWC")))) {
+                    // NB: If user didn't provide input model layout explicitly set it to
+                    // "NCHW" in order to perform conversion to image layout (NHWC).
+                    if (!explicit_in_model_layout) {
+                        input_info.model().set_layout(::ov::Layout("NCHW"));
+                    }
+                    // Image has NHWC layout.
+                    input_info.tensor().set_layout(::ov::Layout("NHWC"))
+                                       .set_spatial_static_shape(matdesc.size.height,
+                                                                 matdesc.size.width);
+                    // NB: Configure resize since image may has
+                    // different size than model expects.
+                    input_info.preprocess()
+                        .resize(::ov::preprocess::ResizeAlgorithm::RESIZE_LINEAR);
+                } else if (isBatchedImage(matdesc)) {
+                    // NB: If user didn't provide input model layout explicitly set it to
+                    // "NCHW" in order to perform conversion to image layout (NHWC).
+                    if (!explicit_in_model_layout) {
+                        input_info.model().set_layout(::ov::Layout("NCHW"));
+                    }
+
+                    auto batch_layout = (matdesc.dims[1] == 3 || matdesc.dims[1] == 1)
+                                        ? ::ov::Layout("NCHW") : ::ov::Layout("NHWC");
+
+                    cv::Size spatial_size = (matdesc.dims[1] == 3 || matdesc.dims[1] == 1)
```

Comment:
or, spatial_size can be obtained from `tensor_layout` if such explicitly specified by user